internal/lints/const_error_decl.go

```diff
@@ -0,0 +1,78 @@
+package lints
+
+import (
+	"go/ast"
+	"go/token"
+
+	tt "github.com/gnolang/tlin/internal/types"
+)
+
+func DetectConstErrorDeclaration(
+	filename string,
+	node *ast.File,
+	fset *token.FileSet,
+	severity tt.Severity,
+) ([]tt.Issue, error) {
+	var issues []tt.Issue
+
+	ast.Inspect(node, func(n ast.Node) bool {
+		genDecl, ok := n.(*ast.GenDecl)
+		if !ok || genDecl.Tok != token.CONST {
+			return true
+		}
+
+		containsErrorsNew := false
+		for _, spec := range genDecl.Specs {
+			valueSpec, ok := spec.(*ast.ValueSpec)
+			if !ok {
+				continue
+			}
+
+			for _, value := range valueSpec.Values {
+				if isErrorNew(value) {
+					containsErrorsNew = true
+					break
+				}
+			}
+			if containsErrorsNew {
+				break
+			}
+		}
+
+		if containsErrorsNew {
+			issue := tt.Issue{
+				Rule:       "const-error-declaration",
+				Filename:   filename,
+				Start:      fset.Position(genDecl.Pos()),
+				End:        fset.Position(genDecl.End()),
+				Message:    "Constant declaration of errors.New() is not allowed",
+				Suggestion: "Use var instead of const for error declarations",
+				Confidence: 1.0,
+			}
```

Comment:
Should apply severity when creating the issue

Reply:
forgot to add https://github.com/gnoverse/tlin/pull/94/commits/f8a1865bf40cf6ebb2f37802a7f7c3c62451298a